### ISSUE TITLE
Protected interface member: disabled implicit implementation causes error "CS8704"

### ISSUE THREAD

**Issue by bernd5** [dotnet/roslyn#43680]
Protected interface members seem to prohibit members with the same name and non explicit interface implementations.

Consider the following code:

```CSharp
using System;

namespace ProtectedInterfaceMember
{
    interface ISomething
    {
        protected void DoInternal()
        {
            Console.WriteLine("Do something which can be changed");
        }

        void Do()
        {
            DoInternal();
        }
    }

    public class App : ISomething
    {
        public void DoInternal()
        {
            Console.WriteLine("This has nothing to do with ISomething");
        }

        public static void Main()
        {
            ISomething app = new App();
            app.Do();
        }
    }
}
```

If you compile it you get error "CS8704", which says you can't override / implement the protected interface member implicitly. If you want to do this it must be done explicit.

I think this should be just a warning or completely removed. If you extend an interface with such protected methods you may break existing code.

In the [spec](https://github.com/dotnet/csharplang/blob/master/proposals/csharp-8.0/default-interface-methods.md) I couldn't find a reason why this situation should cause an error.

**Comment by bernd5** [comment]
Any comments?

**Comment by bernd5** [comment]
see: https://github.com/dotnet/csharplang/discussions/4718

**Comment by CyrusNajmabadi** [comment]
This feels like a bug to me @jaredpar .  Take a look at the convo there.  It seems like adding such member (despite having a DIM impl) in hte interface, can break consumers of the interface.  That seems... bad.

**Comment by bernd5** [comment]
Roslyn seems to produce now fine IL code. But the runtime is currently not spec conforment. Overriding of protected interface members is only allowed via explicit interface implementations. Therefore the expected output is:

```
Do something which can be changed
```

but the current runtime prints:

```
This has nothing to do with ISomething
```

**Comment by bernd5** [comment]
The IL is still not good, because the method `App.DoInternal()` becomes silently `final virtal`.
MichalStrehovsky additionaly observed that the "rename" refactoring renames the non interface related method, too.
